streams/src/main/java/org/apache/kafka/streams/processor/internals/StateDirectory.java

```diff
@@ -347,8 +349,14 @@ private synchronized void cleanRemovedTasks(final long cleanupDelayMs,
      * @return The list of all the existing local directories for stream tasks
      */
     File[] listTaskDirectories() {
```

Comment:
One last thing: Could you open another PR to add unit tests that check that the array is empty for the two edge cases?

Reply:
Ack, added the tests to the Pt 2.5 PR

Reply:
Actually, ended up doing some additional cleanup on the side so I split it out into a  small PR; please give this a quick review.
https://github.com/apache/kafka/pull/8304